### ISSUE TITLE
Fix configuration of theme while locked

### ISSUE THREAD

**Issue by markmhendrickson** [blockstack/blockstack-app#4826]
https://github.com/leather-wallet/extension/assets/28991/90202f76-20d1-4bf1-8a48-832f261f52bb

**Comment by pete-watters** [comment]
This is the same issue as https://github.com/leather-wallet/extension/issues/4783 . 

I can look into it if you want?

**Comment by pete-watters** [comment]
@kyranjamie  - what are your thoughts on making the `Settings Menu` dialogs not routable? 
- change theme
- change network 
- sign out 

I believe this will simplify things and help fix this bug and https://github.com/leather-wallet/extension/issues/4783. 

We can fire an analytics page view inside of the dialogs. 

The account selector is using an un-routed dialog now.

**Comment by kyranjamie** [comment]
Agree. Think we should do this, and remove the `/*` background routing code.

Think we should still route modals when we can, but not for global ones.

**Comment by pete-watters** [comment]
OK. I can fix the settings ones now in #4370 which will solve this bug. 

Then we can attempt to move the `/*` routing seperately

**Comment by pete-watters** [comment]
I'll just fix this as part of https://github.com/leather-wallet/extension/issues/4370. 

It's very similar to https://github.com/leather-wallet/extension/issues/4783 and relates to problems with our routing from the settings menu. 

I needed to rework the settings menu anyway to use the new `DropDown` component. It's not much more effort to implement https://github.com/leather-wallet/extension/issues/4313 , at least partially. 

In the new design, I can make use of the radix dropdown sub-menu to replace the modals for `Theme` and `Change network`

![Screenshot 2024-02-08 at 06 20 02](https://github.com/leather-wallet/extension/assets/2938440/adfc9d55-9d0a-4769-b010-5c989bbee1b3)

Then I will just have to implement a fix to the modal for `Sign Out`

**Comment by pete-watters** [comment]
I added this new menu:
- I used the existing menu items we already have 
- I need to add the new icons

Using the new `radix` dropdown and the submenus aren't working too well. I've decided that for now I will keep using modals for sub-menu items but not have them routable. 

That way we can still fix this issue and https://github.com/leather-wallet/extension/issues/4783 but we don't need to also tackle issues with the submenu.
